client/sites-dashboard/controller.tsx

```diff
@@ -28,14 +29,25 @@ const getStatusFilterValue = ( status?: string ) => {
 	);
 };
 
+export function sanitizeQueryParameters( context: PageJSContext, next: () => void ) {
+	context.query.status = getStatusFilterValue( context.query.status );
+
+	if ( context.query.status === DEFAULT_SITE_LAUNCH_STATUS_FILTER_VALUE ) {
+		const pathWithQuery = window.location.pathname + window.location.search;
+		history.replaceState( null, '', removeQueryArgs( pathWithQuery, 'status' ) );
```

Comment:
Let me know if there's a better way of doing that. I tried to `page.replace` but it loops infinitely. cc @Automattic/build

Reply:
I like that the query arg sanitisation is done in one of the router middleware. Gives me ExpressJS vibes.

You can also do `context.canonicalPath` instead of `window.location.pathname + window.location.search`. With the advantage it won't crash during SSR.

I think the infinite loop is happening because `context.query.status` is being updated _before_ the redirect. Doing it after fixes the loop.

I also added an early so that we don't call `next()`; bailing the middleware chain early since we know we don't want to render anything.

```typescript
export function sanitizeQueryParameters( context: PageJSContext, next: () => void ) {
	const sanitized = getStatusFilterValue( context.query.status );

	if ( sanitized === DEFAULT_SITE_LAUNCH_STATUS_FILTER_VALUE ) {
		context.page.replace( removeQueryArgs( context.canonicalPath, 'status' ) );
		return;
	}

	context.query.status = sanitized;

	next();
}
```

I rebuilt `Node` and half expected to see a `3xx` redirect when I added `?status=all`, but it didn't 🤷

Reply:
Thanks for the suggestion! I tweaked it a bit because it was stuck in an intermediary state between removing the query parameter and not doing anything after that 😅

Now it's SSR-friendly. You can check the implementation [here](https://github.com/Automattic/wp-calypso/pull/66511/commits/6460136fd16efa955c4fbc090b74e04ea77a2ff8) and we can tweak it in a follow-up PR. I'm merging this for now!